### PR TITLE
Add a new e2e test for the build configure page

### PR DESCRIPTION
**Pull request by sbelsk**
This PR adds a new Cypress end-to-end test for the `builds/<buildid>/configure` page, as part of our efforts to increase the test coverage. The changes also include a fix to a typo in the pluralization rerouting logic (introduced in https://github.com/Kitware/CDash/pull/1900) that broke all existing references to the url: `build/<buildid>/configure`.

### CODE REVIEW

File: tests/cypress/e2e/build-configure.cy.js
Comment (williamjallen) [inline tests/cypress/e2e/build-configure.cy.js:9]: Why do we care about verifying that the route matches a given pattern?  It does nothing to verify that the configure page actually lives at that location.  Instead, you could click on the link and expect some text to exist.  That runs into a similar issue in that the test will have to be updated if the text is changed, but at least it guarantees some amount of correctness.

File: tests/cypress/e2e/build-configure.cy.js
Comment (williamjallen) [inline tests/cypress/e2e/build-configure.cy.js:1]: Instead of checking against data generated by the test suite, would it make sense to check against data generated by the DB seeding script I created in #1964?

**Review by williamjallen — CHANGES_REQUESTED** [review]
Please make the redirect bugfix in a separate PR targeting the 3.3 release branch.

**Comment by sbelsk** [comment]
Good notes. I opened PR https://github.com/Kitware/CDash/pull/2016 for the patch fix, which will block this PR until it gets merged.

File: tests/cypress/e2e/build-configure.cy.js
Comment (sbelsk) [reply tests/cypress/e2e/build-configure.cy.js:9]: I have changed this bit to instead test the header of the page which should show `<project> - Configure`

**Review by williamjallen — APPROVED** [review]

**Pull request merged**
